core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java

```diff
@@ -844,6 +844,28 @@ MatchResult detailMatches(PlanNode node, StatsProvider stats, Session session, M
         return match(newAliases.build());
     }
 
+    public <T extends PlanNode> PlanMatchPattern with(Class<T> clazz, Predicate<T> predicate)
```

Comment:
maybe call this `matching`?

Reply:
It's simplification of `with` method below that accepts full `Matcher`. Also `withXXX` seems to be convention in this class.